"Microsoft.Orleans.CodeGenerator.MSBuild" fails to code generate when "*.Orleans.*" occurs in (at a minimum) grains impls or grains interface namespace

In the process of upgrading to .net core 3.1, and orleans 3.0.1, I ran into an issue with "Microsoft.Orleans.OrleansCodeGenerator.Build" code generation - i believe this has happened previously as this generator relies on a specific version of the framework, and has not yet been released since 3.1.0 dnc was released.

I did not realize there was a second code generator, which seems to be independent of the dnc version - "Microsoft.Orleans.CodeGenerator.MSBuild".

I inquired about this on your [gitter](https://gitter.im/dotnet/orleans?at=5deaed7b26eeb8518f7b7435), but here's the problem described in the link:

* Project was previously working on dnc3.0, using "Microsoft.Orleans.OrleansCodeGenerator.Build"
* Update to dnc3.1, "Microsoft.Orleans.OrleansCodeGenerator.Build" fails on code generation with:
```
Microsoft.Orleans.OrleansCodeGenerator.Build.targets(77, 5): [MSB3073] The command ""C:\Program Files\dotnet\dotnet.exe" "C:\Users\rlh4\.nuget\packages\microsoft.orleans.orleanscodegenerator.build\3.0.0\build\..\tasks\netcoreapp3.0\Orleans.CodeGeneration.Build.dll" "@obj\Debug\netcoreapp3.1\codegen\.....Grains.Interfaces.orleans.g.args.txt"" exited with code 3.
```
* Change "Microsoft.Orleans.OrleansCodeGenerator.Build" to "Microsoft.Orleans.CodeGenerator.MSBuild" as suggested by @srberard
* With "Microsoft.Orleans.OrleansCodeGenerator.Build", getting build errors during code generation of `dotnet build`:
```
obj\Debug\netcoreapp3.1\MyNamespace.Orleans.Grains.Interfaces.orleans.g.cs(9436,13): error CS0234: The type or namespace name 'CodeGeneration' does not exist in the namespace 'MyNamespace.Orleans' (are you missing an assembly reference?) [...\MyNamespace.Orleans.Grains.Interfaces\MyNamespace.Orleans.Grains.Interfaces.csproj]
```

This issue ***seems*** to be related to having a namespace in my code `*.Orleans.*`.  I have set up a repro of the issue [here](https://github.com/Kritner-Blogs/OrleansGettingStarted/pull/17/commits/0fcbe07c7a6c3fb824bd49036d43f560e7f0fb23).  Note that in the previous [commit](https://github.com/Kritner-Blogs/OrleansGettingStarted/pull/17/commits/b3b77d3daa1542303406e1b85e95476748a47a11), the code generation worked, prior to my projects containing `*.Orleans.*` in their namespaces.

Without `*.Orleans.*` in namespace:
![image](https://user-images.githubusercontent.com/8814983/70435637-5b07c680-1a55-11ea-982c-b4dd04b70c92.png)


With `*.Orleans.*` in namespace:
![image](https://user-images.githubusercontent.com/8814983/70435536-17ad5800-1a55-11ea-9c63-461f5c436cd3.png)

The runnable projects for this solution are:

* Kritner.OrleansGettingStarted.SiloHost
* Kritner.OrleansGettingStarted.Client

both of which can be run with the basic `dotnet run` from their respective project directories (though note the interfaces and grains projects fail to build on the most up to date commit on that branch... https://github.com/Kritner-Blogs/OrleansGettingStarted/pull/17

--- a/src/Orleans.CodeGenerator/Generators/GrainReferenceGenerator.cs
+++ b/src/Orleans.CodeGenerator/Generators/GrainReferenceGenerator.cs
@@ -134,7 +134,7 @@ namespace Orleans.CodeGenerator.Generators
                     {
                         body.Add(
                             ExpressionStatement(
-                                InvocationExpression(wellKnownTypes.GrainFactoryBase.ToDisplayString().ToIdentifierName().Member("CheckGrainObserverParamInternal"))
+                                InvocationExpression(wellKnownTypes.GrainFactoryBase.ToNameSyntax().Member("CheckGrainObserverParamInternal"))
                                     .AddArgumentListArguments(Argument(parameter.Name.ToIdentifierName()))));
                     }
                 }

